json_test.go

```diff
@@ -41,3 +43,106 @@ func TestJsonUnmarshalQuery(t *testing.T) {
 		return
 	}
 }
+
+type JsonValues struct {
+	Text      string      `json:"text"`
+	Integer   int         `json:"integer"`
+	Decimal   float64     `json:"decimal"`
+	Bool      bool        `json:"bool"`
+	Array     []any       `json:"array"`
+	SubObject *JsonValues `json:"obj"`
+}
+
+func TestJsonUnmarshalError(t *testing.T) {
+	var (
+		target          JsonValues
+		unmarshalErr    ErrJsonUnmarshal
+		unmarshalErrPtr *ErrJsonUnmarshal
+	)
+
+	type testCase struct {
+		rawJson                    string
+		expectedSourceType         string
+		expectedTargetType         string
+		expectedTargetPropertyName string
```

Comment:
In kurzen funcs braucht man keine langen Variablen-Namen. Gerade in diesem fall heir würde ich mit kürzeren Identifiern arbeiten.